server.js

```diff
@@ -1072,6 +1073,80 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Liberapay integration.
+camp.route(/^\/liberapay\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var type = match[1];  // e.g., 'gives'
+  var entity = match[2]; // e.g., 'Changaco'
+  var format = match[3];
+  var apiUrl = 'https://liberapay.com/' + entity + '/public.json';
+  // Establish default
+  var badgeData = getBadgeData('receives', data);
+  switch(type) {
+      case 'gives':
+          badgeData = getBadgeData('gives', data);
+          break;
+      case 'patrons':
+          badgeData = getBadgeData('patrons', data);
+          break;
+      case 'goal':
+          badgeData = getBadgeData('goal progress', data);
+          break;
+      }
```

Comment:
How about:
```js
const label = {
  receives,
  gives,
  patrons,
  goal: 'goal progress',
}[type];
const badgeData = getBadgeData(label, data);
```

Seems like it would be a good idea to lock down the `type` part of the regex to accept only these strings. That way you don't need to handle the errors.

Reply:
Sounds good. Fixed in https://github.com/badges/shields/pull/1251/commits/47803f78bd7e915c40583739e9474cd964981876.

Reply:
Could you edit the regex too, so it only accepts the four acceptable values?

Reply:
Weird I didn't see this. Think I fixed in https://github.com/badges/shields/pull/1251/commits/f0a84f758fb34676b7ebd70898f28dbcbb8abadb.